demo/admin/src/pages/PageContentBlock.tsx

```diff
@@ -46,3 +46,11 @@ export const PageContentBlock = createBlocksBlock({
         return <UserGroupChip item={item} />;
     },
 });
+
+const orginalCreatePreviewState = PageContentBlock.createPreviewState;
+
+PageContentBlock.createPreviewState = (state, context: IPreviewContext) => {
+    const previewState = orginalCreatePreviewState(state, context);
+
+    return { ...previewState, blocks: previewState.blocks.filter((block) => block.userGroup === context.userGroup || block.userGroup === "All") };
+};
```

Comment:
This isn't ideal and we need to come up with a better solution. The rest is rather straightforward IMO.